### ISSUE TITLE
Extend Metrics: up / version

### ISSUE THREAD

**Issue by strowi** [vmware-tanzu/velero#7463]
**Describe the problem/challenge you have**

Currently velero isn't exporting a prometheus-metrics regarding the status of the velero-pod. This makes it harder to check if it is currently running. E.g. if the pod is not scheduled or crashing all i can do is an alert on the last time of a successful backup which might be delayed or not the correct one.

**Describe the solution you'd like**

Add a simple metric like `velero_up` or `velero_build_info` with its status or version-number.

**Anything else you would like to add:**

v

**Environment:**

- Velero version (use `velero version`): 1.13.0
- Kubernetes version (use `kubectl version`): 1.27.10-k3s
- Kubernetes installer & version: 
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by blackpiglet** [comment]
In normal cases, the Velero pod can update the proposed `velero_up`metric when it exits, but for some cases, e.g. OOM, the Velero pod may not have the chance to update the metric.

Does watching the selected pod-killed event sound reasonable for your case?

**Comment by m-messiah** [comment]
velero_up sounds like duplicate of normal prometheus `up` metric and for the other pod statuses consider looking at https://github.com/kubernetes/kube-state-metrics or similar project.

velero_build_info on the other hand would be a very nice thing to track rollouts. Is there any specific reason why is it not here yet?

**Comment by blackpiglet** [comment]
@m-messiah 
By saying `up` metric, do you mean metric like this `kube_deployment_status_replicas_ready`?

**Comment by m-messiah** [comment]
> [@m-messiah](https://github.com/m-messiah) By saying `up` metric, do you mean metric like this `kube_deployment_status_replicas_ready`?

No, I mean `up{job="velero"}` that is autogenerated by prometheus and shows the status of scraping: https://prometheus.io/docs/concepts/jobs_instances/#automatically-generated-labels-and-time-series

It could be easily added as an alert like `absent(up{job="velero"})` or `sum(up{job="velero"} or vector(0)) < [replicas]`

KSM metrics like `kube_deployment_status_replicas_ready` or `kube_pod_container_status_running` or `kube_pod_status_ready` could be also used, but might be an overkill.
